x-pack/plugins/apm/server/lib/service_map/get_service_map_service_node_info.ts

```diff
@@ -52,20 +58,30 @@ export async function getServiceMapServiceNodeInfo({
   };
 
   const [
+    environmentsWithData,
     errorMetrics,
-    transactionMetrics,
+    transactionKPIs,
     cpuMetrics,
     memoryMetrics,
   ] = await Promise.all([
+    getAllEnvironments({
+      serviceName,
+      setup,
+      includeMissing: true,
+    }),
     getErrorMetrics(taskParams),
-    getTransactionMetrics(taskParams),
+    getTransactionKPIs(taskParams),
     getCpuMetrics(taskParams),
     getMemoryMetrics(taskParams),
   ]);
 
+  const hasEnvironmentData =
+    !environment || environmentsWithData.includes(environment);
+
```

Comment:
Same: is this still used?

Reply:
we can remove this and just look to see if any data was returned instead.

Reply:
Yeah, we shouldn't need to do the `getAllEnvironments` lookup